public/scripts/openai.js

```diff
@@ -3093,12 +3100,43 @@ async function onModelChange() {
         if (value && (value.includes('claude') || value.includes('palm-2'))) {
             oai_settings.temp_openai = Math.min(claude_max_temp, oai_settings.temp_openai);
             $('#temp_openai').attr('max', claude_max_temp).val(oai_settings.temp_openai).trigger('input');
-        }
-        else {
+        } else {
             oai_settings.temp_openai = Math.min(oai_max_temp, oai_settings.temp_openai);
             $('#temp_openai').attr('max', oai_max_temp).val(oai_settings.temp_openai).trigger('input');
         }
 
+        // Get meta-information provided by OpenRouter
+        const modelMetaInformation = model_list.find((model) => model.id?.toLowerCase() === oai_settings.openrouter_model?.toLowerCase());
+        const statusContainer = document.getElementById('openrouter_instruct_status');
+        const errorContainer = document.getElementById('openrouter_instruct_error');
+
+        statusContainer.textContent = '';
+        errorContainer.textContent = '';
+
+        if (true === power_user.instruct.enabled &&
+            true === oai_settings.openrouter_instruct_autoselect &&
+            modelMetaInformation.architecture.instruct_type) {
+            const presetSelect = document.getElementById('instruct_presets');
+
+            const preset = getDefaultPresetForInstructType(modelMetaInformation.architecture.instruct_type);
+            const presetOption = presetSelect.querySelector(`option[value="${preset}" i]`);
```

Comment:
The query will match <option value="Alpaca" />, <option value="alpaca" />, <option value="AlPACa" /> and so on thanks to the case-insensitivity qualifier.

I could, but since the in-memory list and the ui should be in sync and i need the option anyway to enable it, i just check there.